Tests/Listener/RestListenerLocaleTest.php

```diff
@@ -57,25 +57,30 @@ public function testCorrectLocaleIsBeingSet(array $acceptedLocales, $expected, $
     public function dataProviderForTestLocaleIsBeingSet()
     {
         return [
-            'select requested locale' => [
+            'select requested locale from the provided configuration' => [
                 ['lt', 'ru'],
                 'lt',
-                'lt'
+                'lt',
             ],
-            'select locale with the highest weight' => [
+            'if no configuration is provided, don\'t change the locale' => [
                 [],
-                'zh',
-                'zh, en-us; q=0.8, en; q=0.6',
+                'en',
+                'fr-CH, fr;q=0.9, lt;q=0.8, de;q=0.7',
             ],
-            'select locale from the defined locale array' => [
+            'select locale from the defined locale configuration despite it having lowest weight' => [
```

Comment:
This test could be improved as currently we use the same locale as the default one - it might be that it's not changed after all.

Reply:
Also the comment is slightly wrong - here the locale is selected from the en-us part, not the en part with lower priority.